src/video/emscripten/SDL_emscriptenevents.c

```diff
@@ -300,25 +300,33 @@ int
 Emscripten_HandleMouseMove(int eventType, const EmscriptenMouseEvent *mouseEvent, void *userData)
 {
     SDL_WindowData *window_data = userData;
-    int mx = mouseEvent->canvasX, my = mouseEvent->canvasY;
+    int mx, my;
+    static double residualx = 0, residualy = 0;
     EmscriptenPointerlockChangeEvent pointerlock_status;
 
+    /* rescale (in case canvas is being scaled)*/
+    double client_w, client_h, xscale, yscale;
+    emscripten_get_element_css_size(NULL, &client_w, &client_h);
+    xscale = window_data->window->w / (client_w * window_data->pixel_ratio);
```

Comment:
I don't think one needs to multiply by `window_data->pixel_ratio` in the denominator here? This is because `mouseEvent->movementX/Y` are specified in CSS pixels already, and `client_w` is also in CSS pixels, so `mouseEvent->movementX/client_w` remaps the coordinate to unitary [0,1] range, and multiplying that by `window_data->window->w` (which is presumably here the WebGL render target size in hardware pixels) gets the coordinate to [0, renderTargetSizeInHardwarePixels] range.